community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -398,6 +398,36 @@ public IndexImplementation apply( String name )
         this.pageCache = pageCache;
     }
 
+    /**
+     * Custom IO integrations are partially supported, but the features below are not functional yet.
+     * When those features are available, this check can be removed entirely.
+     * This will allow us to ship the custom IO integration for experimental use, and protect users from
+     * using it with the currently unsupported features. i.e using custom IO integration in HA mode.
+     */
+    private Config validateSettingsForPageSwapper( Config config )
+    {
+        if ( config.get( GraphDatabaseSettings.pagecache_swapper ) != null )
+        {
+            if ( config.get( GraphDatabaseSettings.allow_store_upgrade ) )
+            {
+                throw new IllegalArgumentException( "Store upgrade not allowed with custom IO integrations" );
+            }
+            String onlineBackup = config.getParams().get( "dbms.backup.enabled" );
```

Comment:
Can we instead to validation in `EditionModule`s? This will allow to use actual enterprise settings and not hardcode them as strings.